src/shared/components/plots/WaterfallPlot.tsx

```diff
@@ -0,0 +1,545 @@
+                d.searchindicatorx = d.order;
+                d.searchindicatory = labelPos;
+            }
+
+        });
+
+        return searchLabels;
+    }
+    
+    @bind
+    private getChart() {
+        return (
+            <div
+                ref={this.containerRef}
+                style={{width: this.svgWidth, height: this.svgHeight}}
+            >
+                <svg
+                    id={this.props.svgId || ""}
+                    style={{
+                        width: this.svgWidth,
+                        height: this.svgHeight,
+                        pointerEvents: "all"
+                    }}
+                    height={this.svgHeight}
+                    width={this.svgWidth}
+                    role="img"
+                    viewBox={`0 0 ${this.svgWidth} ${this.svgHeight}`}
+                    >
+                    <g
+                        transform={`translate(${LEFT_PADDING},0)`}
+                        >
+                        <VictoryChart
+                            theme={CBIOPORTAL_VICTORY_THEME}
+                            width={this.props.chartWidth}
+                            height={this.props.chartHeight}
+                            standalone={false}
+                            domainPadding={PLOT_DATA_PADDING_PIXELS}
+                            singleQuadrantDomainPadding={false}
+                        >
+                            {this.title}
+                            {this.legend}
+                            {this.props.horizontal && <VictoryAxis
+                                domain={this.plotDomainX}
+                                orientation="bottom"
+                                offsetY={50}
+                                crossAxis={false}
+                                tickCount={NUM_AXIS_TICKS}
+                                tickFormat={this.tickFormatX}
+                                axisLabelComponent={<VictoryLabel dy={25}/>}
+                                label={this.props.axisLabel}
+                            />}
+                           {!this.props.horizontal && <VictoryAxis
+                                domain={this.plotDomainY}
+                                offsetX={50}
+                                orientation="left"
+                                crossAxis={false}
+                                tickCount={NUM_AXIS_TICKS}
+                                tickFormat={this.tickFormatY}
+                                dependentAxis={true}
+                                axisLabelComponent={<VictoryLabel dy={-35}/>}
+                                label={this.props.axisLabel}
+                            />}
+                            <VictoryBar
+                                // barRatio={1} // removes spaces between bars
+                                style={{
+                                    data: {
+                                        fill: (d:D) => d.fill,
```

Comment:
it's significantly more efficient with Victory to make a separate VictoryBar element for each style settings, rather than using functions. You can see an example of how I do this in `ScatterPlot.tsx` in the getChart() method with VictoryScatter elements

Reply:
In the case of scatterplot the approach is to partition the datapoints into groups of identical styling and then consecutively plot different groups of points with hard coded styling information. For bar plots such as the waterfallplot this is not possible because the bar widths and inter-bar widths should be evaluated with all bars being present. When taking the scatterplot approach leads to malformed graphs. This was in fact my first implementation that I had to change to make it work.

Reply:
Can you calculate the bar widths using all the data together, then separate them out for input to victory?

Reply:
Maybe, I don't know. But how much of a problem is the current implementation? Will this performance difference be noticeable by the user? To be honest we do not have budget left to explore alternative ways that may or may not work, and I suspect that your proposed change is not guaranteed to be successful . Can't we not postpone this mod to a later time point?